tsdb/engine/tsm1/cache.go

```diff
@@ -455,8 +455,20 @@ func (c *Cache) write(key string, values []Value) {
 }
 
 func (c *Cache) entry(key string) *entry {
-	c.mu.Lock()
+	// low-contention path: entry exists, no write operations needed:
+	c.mu.RLock()
 	e, ok := c.store[key]
+	if ok {
+		c.mu.RUnlock()
```

Comment:
as c.mu.RUnlock is invoked for both branches, wouldn't it make sense to instead invoke it immediately before the branch? Or is this a go idiom i am unfamiliar with.